packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -98,8 +100,26 @@ class Route extends EmberObject implements IRoute {
   controller!: Controller;
   currentModel: unknown;
 
+  _bucketCache!: BucketCache;
   _internalName!: string;
   _names: unknown;
+  _router!: EmberRouter;
+
+  constructor(owner: Owner) {
+    super(...arguments);
+
+    if (owner) {
+      let router = owner.lookup<EmberRouter>('router:main');
+      let bucketCache = owner.lookup<BucketCache>(P`-bucket-cache:main`);
+
+      assert('Expected route injections to be defined', router && bucketCache);
```

Comment:
This assertion isn't really helpful messaging wise, given this message I would have no clue what it is trying to say...

Reply:
I agree, I wasn't really meaning these assertions to be for users. They were mainly for types, and for us for internal development purposes. Since these were previously `!` assertions, my assumption was that these couldn't be issues.

I updated the assertions to be more explicit that these are internal bugs.